### PR TITLE
[RHOSINFRA-159] - Upgrade cloud from 7.3 to 8.0

### PR DESCRIPTION
**Pull request by ovorobio**

### CODE REVIEW

**Comment by yfried-redhat** [comment]
retest openstack

**Comment by yfried-redhat** [comment]
@cmyster please review

File: playbooks/installer/ospd/upgrade/7_3-8_0.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
+---
+- name: Prepare the undercloud
+  hosts: undercloud
+  gather_facts: no
+  tasks:
+    - name: Setting up node
+      yum: name=http://sat618-qe-rhel7.usersys.redhat.com/pub/katello-ca-consumer-latest.noarch.rpm state=present
```
Comment (tkammer) [inline playbooks/installer/ospd/upgrade/7_3-8_0.yml]: Note, any internal URLs should be in private.yml, not explicit here.
Also, format:
`yum:
    name: <URL>
    state: present
`

File: playbooks/installer/ospd/upgrade/7_3-8_0.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    - name: register subscription manager
+      shell: 'subscription-manager register --org="Default_Organization" --activationkey="ak-osp8"'
```
Comment (tkammer) [inline playbooks/installer/ospd/upgrade/7_3-8_0.yml]: Same with subscription manager, need to be private.yml

File: playbooks/installer/ospd/upgrade/7_3-8_0.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      ignore_errors: yes
+
+    - name: print the last few lines of the output to spot failures
+      shell: "tail -n30 undercloud_upgrade.log"
+
+    - fail: msg="Cloud upgrade failed... :("
+      when: undercloud_upgrade.rc != 0
+
+    - name: Setting up node
+      shell: "source ~/stackrc; for i in `nova list|awk '/ACTIVE/ {print $(NF-1)}'|awk -F\"=\" '{print $NF}'`;\
```
Comment (tkammer) [inline playbooks/installer/ospd/upgrade/7_3-8_0.yml]: All these are basically rewriting ANSIBLE(!).
This should be:
```
hosts: overcloud_nodes
tasks:
    - name: subscribe to RHN
      shell: .....
```

File: playbooks/installer/ospd/upgrade/7_3-8_0.yml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                sudo crudini --set /etc/yum.repos.d/redhat.repo rhel-7-server-rpms gpgcheck 0'; done"
+
+    - name: get templates imported during deployment
+      shell: "cat ~/overcloud_deploy.sh | grep my_templates"
+      register: templates
+
+    - name: create script for initiating pacemaker upgrade
+      template:
+        src: overcloud_upgrade.sh.j2
+        dest: "~/overcloud_upgrade_1.sh"
```
Comment (tkammer) [inline playbooks/installer/ospd/upgrade/7_3-8_0.yml]: Why not "overcloud_upgrade.sh"?

**Comment by tkammer** [comment]
:-1: 
I understand that this is a simple "copy & paste" from the bash script... but if you're converting this to Ansible, please follow Ansible guidelines.
Most of the tasks there should be Ansible modules and not shell.
Lets sync about this and fix what needs to be fixed.

**Pull request closed without merging**